### PR TITLE
CBL-5803 : Implement Lazy Vector Index and Test for Swift

### PR DESCRIPTION
**Pull request by pasin**
* Implemented Lazy Vector Index and Test for Swift.
* Fixed missing symbols.
* Refactored VectorSearchTest in both Swift and Objective-C to reduce duplicate codes.
* Updated some API docs.
* The companion change commit is in the EE repo.

### CODE REVIEW

File: Swift/Tests/VectorSearchTest.swift [hunk 1]
```diff
@@ -701,28 +688,12 @@ class VectorSearchTest: CBLTestCase {
...
-        let explain = try q.explain() as NSString
-        XCTAssertNotEqual(explain.range(of: "SCAN kv_.words:vector:words_index").location, NSNotFound)
-        
-        let rs = try q.execute()
+        let rs = try executeWordsQuery(limit: 20)
         XCTAssertEqual(rs.allResults().count, 20)
-        XCTAssert(checkIndexWasTrained())
     }
     
     /// FAILED : https://issues.couchbase.com/browse/CBL-5538
```
Comment (velicuvlad) [inline Swift/Tests/VectorSearchTest.swift:699]: Can we re-enable 12 bits?

File: Swift/Tests/VectorSearchTest.swift [hunk 2]
```diff
@@ -753,35 +724,18 @@ class VectorSearchTest: CBLTestCase {
     ///     10. Reset the custom logger.
     ///     11. Repeat steps 2 to 10 by changing the PQ’s bits to 4 and 12 respectively.
     func testCreateVectorIndexWithPQ() throws {
-        let collection = try! db.collection(name: "words")!
-        
         for numberOfBits in [8, 4, /*12*/] {
```
Comment (velicuvlad) [inline Swift/Tests/VectorSearchTest.swift:727]: here

**Review by velicuvlad — APPROVED** [review]

**Review by bmeike — APPROVED** [review]

File: Swift/Tests/VectorSearchTest.swift (hunk 1 above)
Comment (pasin) [reply Swift/Tests/VectorSearchTest.swift:699]: Yes. Can you do that when go over all of the failed tests?

File: Swift/Tests/VectorSearchTest.swift [hunk 3]
```diff
@@ -807,32 +761,27 @@ class VectorSearchTest: CBLTestCase {
...
+            try deleteWordsIndex()
             config.encoding = .productQuantizer(subquantizers: UInt32(numberOfSubq), bits: 8)
             expectExcepion(exception: .invalidArgumentException) {
-                try! collection.createIndex(withName: "words_index", config: config)
+                try? self.createWordsIndex(config: config)
             }
         }
     }
     
-    /// https://issues.couchbase.com/browse/CBL-5537
```
[the remark this answers is not in the document]
Comment (pasin) [reply Swift/Tests/VectorSearchTest.swift:835]: We will use centroids = 8 for our tests. We improved the warning message instead.

**Pull request merged**